src/server/internalServer.cpp

```diff
@@ -692,28 +709,24 @@ std::unique_ptr<Response> InternalServer::handle_catalog(const RequestContext& r
       startIndex = extractFromString<unsigned long>(request.get_argument("start"));
     } catch (...) {}
     try {
-      filter.acceptTags(kiwix::split(request.get_argument("tag"), ";"));
+      tags = kiwix::split(request.get_argument("tag"), ";");
+    } catch (...) {}
+    try {
+      tags.push_back("_category:" + request.get_argument("category"));
```

Comment:
If the book now has a `getCategory`, we should use it (as a new method in `Filter`) instead of reusing the `acceptTags`.